csharp/src/Ice/Reference.cs

```diff
@@ -584,6 +584,101 @@ static void StartQueryOption(StringBuilder sb, ref bool firstOption)
             }
         }
 
+        internal async ValueTask<Connection> GetConnectionAsync(CancellationToken cancel)
+        {
+            (Connection? connection, bool cached, IReadOnlyList<Endpoint> endpoints, List<Connector>? connectors) =
+                await GetExistingConnectionAsync(cancel).ConfigureAwait(false);
+
+            if (connection == null)
+            {
+                OutgoingConnectionFactory connectionFactory = Communicator.OutgoingConnectionFactory;
+                if (connectors == null)
+                {
+                    (cached, endpoints) = await ComputeEndpointsAsync(cancel).ConfigureAwait(false);
```

Comment:
This is a bit confusing, the "if" checks for connectors but not for endpoints and then we compute new endpoints. Can't we actually remove the endpoint/connector computation here and fix GetExistingConnection to always return computed endpoints/connectors when if can't find an existing connection? It would be simpler and avoid doing the computation here.

Reply:
I removed `GetExistingConnection` as per your other suggestion